packages/grid_client/tests/modules/gateways.test.ts

```diff
@@ -185,55 +233,174 @@ test("TC1237 - Gateways: Expose a VM Over Gateway", async () => {
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(gatewayNodeId);
+
+  gatewayResult = await gridClient.gateway.getObj(gw1.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name + "mycelium");
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  gatewayAsseration(gatewayResult);
+
+  //--------------------IPv6--------------------
+  backends = ["http://[" + result[0].publicIP["ip6"].split("/")[0] + "]:8000"];
+  log(backends);
+
+  //Name Gateway Model
+  const gw2: GatewayNameModel = {
+    name: name + "ipv6",
+    node_id: gatewayNodeId,
+    tls_passthrough: tlsPassthrough,
+    backends: backends,
+  };
+
+  gatewayRes = await gridClient.gateway.deploy_name(gw2);
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(gatewayNodeId);
+
+  gatewayResult = await gridClient.gateway.getObj(gw2.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name + "ipv6");
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  gatewayAsseration(gatewayResult);
+
+  //--------------------IPv4--------------------
+  backends = ["http://" + result[0].publicIP["ip"].split("/")[0] + ":8000"];
+  log(backends);
+
+  //Name Gateway Model
+  const gw3: GatewayNameModel = {
+    name: name + "ipv4",
+    node_id: gatewayNodeId,
+    tls_passthrough: tlsPassthrough,
+    backends: backends,
+  };
+
+  gatewayRes = await gridClient.gateway.deploy_name(gw3);
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(gatewayNodeId);
+
+  gatewayResult = await gridClient.gateway.getObj(gw3.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name + "ipv4");
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  gatewayAsseration(gatewayResult);
```

Comment:
These lines are common in all gateway tests. Maybe we can remove it to a separate function and use it here?

ex. async function testGateway(name: string){
   all logic 
}